src/lib.rs

```diff
@@ -1,3 +1,66 @@
-#[test]
-fn it_works() {
+mod test;
+
+use std::collections::HashMap;
+
+/// A structure containing all the common attributes
+/// of the basic OSM data types
+/// 
+/// See: [OSM wiki - Common attributes](https://wiki.openstreetmap.org/wiki/Elements#Common_attributes)
+pub struct ElementInfo {
+    pub id: i64,
+    pub user: String,
+    pub uid: i32,
```

Comment:
The user fields should be derived from the uid field, no?

Reply:
OSM users can change their username, so the user field will (I think?) be the name they used at the time. uids won't changed. See below for optional user fields